specification/network/resource-manager/Microsoft.Network/stable/2019-08-01/virtualWan.json

```diff
@@ -2529,45 +2601,210 @@
+            "schema": {
+              "$ref": "#/definitions/P2SVpnGateway"
+            }
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          }
+        },
+        "x-ms-examples": {
+          "P2SVpnGatewayGetConnectionHealth": {
+            "$ref": "./examples/P2SVpnGatewayGetConnectionHealth.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/p2svpnGateways/{gatewayName}/getP2sVpnConnectionHealthDetailed": {
+      "post": {
+        "tags": [
+          "P2SVpnGateways"
+        ],
+        "operationId": "P2sVpnGateways_GetP2sVpnConnectionHealthDetailed",
+        "description": "Gets the sas url to get the connection health detail of P2S clients of the virtual wan P2SVpnGateway in the specified resource group.",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },		
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "name": "gatewayName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the P2SVpnGateway."
+          },
+          {
+            "name": "request",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/P2SVpnConnectionHealthRequest"
+            },
+            "description": "Request parameters supplied to get p2s vpn connections detailed health."
+          },		  
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "P2S connections detailed health report.",
+            "schema": {
+              "$ref": "#/definitions/P2SVpnConnectionHealth"
+            }
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          }
+        },
+        "x-ms-examples": {
+          "P2SVpnGatewayGetConnectionHealthDetailed": {
+            "$ref": "./examples/P2SVpnGatewayGetConnectionHealthDetailed.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualWans/{virtualWANName}/vpnServerConfigurations": {
```

Comment:
>vpnServerConfigurations [](start = 129, length = 23)

The name `vpnServerConfigurations` suggests that this is a nested resource type but this is an action on the wan. Can you rename it to something more imperative?

Reply:
No as per new design. VpnServerConfiguration is a new Top level resource. It contains the VPN configurations which resides on server (P2SVpnGateway) when attached to it. Actually, we have finalized this resource name after discussing throughly.